### ISSUE TITLE
CSRF token mismatch using connect-pg-simple

### ISSUE THREAD

**Issue by bryan** [iaincollins/nextjs-starter#19]
Thanks for iterating upon the latest version of next. It's nice to see the changes through your project.

I've actually been using an [older committed version of this repo version](https://github.com/iaincollins/nextjs-starter/tree/f972f73ff8e407934aab8f078b6c0c460a037669) and can't seem to get other types of compatible session stores to work.

For example, changing the `store` from FileStore to [https://www.npmjs.com/package/connect-pg-simple](connect-pg-simple) seems to always bring up CSRF mismatch, and I'm not quite sure where to start looking to come to a solution. 

Any direction or help would be appreciated!
Thanks

**Comment by iaincollins** [comment]
Hi @bryan - Oh sorry to hear that doesn't work well, thanks for the issue.

I confess I haven't actually tried it with store values other than FileStore! I was assuming that would be abstracted away by the express-sessions and the csrf lib. I'm away for 2-3 days now but will try it out with some other stores when I get back. In the meantime might be worth looking to see if there are some other stores that support Postgres and seeing if they work any better?

For reference the only places it touches are [components/session.js](https://github.com/iaincollins/nextjs-starter/blob/master/components/session.js) (note: this runs on both the client and server) and in the express route setup at [routes/auth.js](https://github.com/iaincollins/nextjs-starter/blob/master/routes/auth.js) so those would be the places to do debugging.

Verifying that  "req.connection._httpMessage.locals._csrf" is populated in Session when running on the server might help. Also logging the output of "res.locals._csrf" might be helpful.

Will take a look when I get back! If the code is inadvertently doing something that's not in practice compatible with different stores I'd definitely want to address that.

**Comment by bryan** [comment]
@iaincollins Thanks for the quick response!

I've tried other PostgreSQL stores including [connect-session-sequelize](https://github.com/mweibel/connect-session-sequelize) but to no avail - running into the same problem. I've also checked to see that the CSRF is populated in `req.connection._httpMessage.locals._csrf` - which it is and the server csrf matches with the client csrf and the `req.csrfToken()` generator is successfully matching as well. 

Not quite sure what exactly is going on or how to proceed since it's not my expertise, but I will await for your input in a couple days.
